src/main/java/io/luna/net/msg/in/CastOnPlayerMessageReader.java

```diff
@@ -0,0 +1,57 @@
+package io.luna.net.msg.in;
+
+import io.luna.game.action.InteractionAction;
+import io.luna.game.event.Event;
+import io.luna.game.event.impl.CastOnPlayerEvent;
+import io.luna.game.model.mob.MobList;
+import io.luna.game.model.mob.Player;
+import io.luna.net.codec.ByteOrder;
+import io.luna.net.codec.ValueType;
+import io.luna.net.msg.GameMessage;
+import io.luna.net.msg.GameMessageReader;
+
+import java.util.Optional;
+
+/**
+ * A {@link GameMessageReader} that generates and submits a {@link CastOnPlayerEvent} when a player attempts to cast
+ * any spell onto
+ * another player.
+ *
+ * @see io.luna.game.action.ActionManager#submit
+ */
+public final class CastOnPlayerMessageReader extends GameMessageReader {
+
+    @Override
+    public Event read(Player caster, GameMessage msg) {
+        int targetId = msg.getPayload().getShort(false, ValueType.ADD);
+        int spellId = msg.getPayload().getShort(false, ByteOrder.LITTLE);
+
+        Optional<Player> targetPlayer = findTarget(caster, targetId);
+        if (targetPlayer.isPresent()) {
+            Player target = targetPlayer.get();
+            this.submitEventFor(target, new CastOnPlayerEvent(caster, spellId, target));
+        }
+        return null;
```

Comment:
Just a note, I think I'll add a specialized "delayed" event type for situations like these. As this is required in a lot of packets